### PR TITLE
LibWeb: Add origin property to window object

### PR DESCRIPTION
**Pull request by Alexander1994**
Added origin property to the window object

Just a small commit / PR to get the hang of making contributions.

### CODE REVIEW

**Comment by timschumi** [comment]
For the future: You can just keep the same PR open and force-push the branch. :^)

File: Userland/Libraries/LibWeb/Bindings/WindowObject.cpp [hunk 1]
```diff
@@ -650,6 +651,17 @@ JS_DEFINE_NATIVE_FUNCTION(WindowObject::screen_y_getter)
     return JS::Value(impl->screen_y());
 }
 
+JS_DEFINE_NATIVE_FUNCTION(WindowObject::origin_getter)
```
Comment (linusg) [inline Userland/Libraries/LibWeb/Bindings/WindowObject.cpp:654]: Please add a spec link above the function: `https://html.spec.whatwg.org/multipage/webappapis.html#dom-origin`

File: Userland/Libraries/LibWeb/Bindings/WindowObject.cpp [hunk 2]
```diff
@@ -650,6 +651,17 @@ JS_DEFINE_NATIVE_FUNCTION(WindowObject::screen_y_getter)
...
+{
+    auto* impl = TRY(impl_from(vm, global_object));
+    auto origin = impl->associated_document().origin();
+    String result = "null";
+    if (!origin.host().is_empty()) {
+        result = String::formatted("{}://{}", origin.protocol(), origin.host());
+    }
```
Comment (linusg) [inline Userland/Libraries/LibWeb/Bindings/WindowObject.cpp:661]: This needs to be abstracted into `Origin::serialize()`, following the spec: https://html.spec.whatwg.org/multipage/origin.html#ascii-serialisation-of-an-origin

**Review by linusg — CHANGES_REQUESTED** [review]

**Comment by paulwratt** [comment]
This should probablyt be checked against the `window.localStorage` (https://github.com/SerenityOS/serenity/commit/47979996e832bdd05440c4a9437920863543441d) where a hash is created against the origin, and weather it affect `window.sessionStorage` in its current form (#12698)

**Comment by Alexander1994** [comment]
> This should probably be checked against the `window.localStorage` ([4797999](https://github.com/SerenityOS/serenity/commit/47979996e832bdd05440c4a9437920863543441d)) where a hash is created against "origin" in [Stroage](https://github.com/SerenityOS/serenity/commit/47979996e832bdd05440c4a9437920863543441d#diff-cc430c8292bfa7b11143fa0e4b0d7ed2277519ae5db85f763551e9fb70999ff8), and weather it affects `window.sessionStorage` in its current form (#12698)

I'm not sure we'd want these to be equivalent origin

**Comment by paulwratt** [comment]
Specifically I meant this (appologies it was in `Window.cpp` not `Storage.cpp`):
```
    static HashMap<Origin, NonnullRefPtr<HTML::Storage>> local_storage_per_origin;
    return local_storage_per_origin.ensure(associated_document().origin(), [] {
        return HTML::Storage::create();
    });
```

**Comment by Alexander1994** [comment]
> Specifically I meant this (appologies it was in `Window.cpp` not `Storage.cpp`):
> 
> ```
>     static HashMap<Origin, NonnullRefPtr<HTML::Storage>> local_storage_per_origin;
>     return local_storage_per_origin.ensure(associated_document().origin(), [] {
>         return HTML::Storage::create();
>     });
> ```

 I opted to create a separate function for providing a hash of the origin vs how we would hash it for a hashmap. I'll have a look soon to confirm though. Thanks for the heads up!

**Comment by Alexander1994** [comment]
A quick check on chrome confirmed that these hash functions should be the same as "file://" has the same origin. I'll update this PR accordingly.

**Review by davidot — COMMENTED** [review]
Code LGTM, just one comment about spec comments

**Review by davidot — APPROVED** [review]

**Review by linusg — APPROVED** [review]

**Pull request merged**